core/utils/files.go

```diff
@@ -170,3 +175,14 @@ func (s FileSize) String() string {
 	str, _ := s.MarshalText()
 	return string(str)
 }
+
+func MakeTestFile(t *testing.T, contents any, fileName string) string {
```

Comment:
Does this need to be a general util func? Can we limit it to tests instead? Or even a toml config test package?
I think we should document the expectations with secrets too, so this isn't mistakenly assumed to be able to marshal secret values.
```suggestion
func WriteTOMLFile(t *testing.T, contents any, fileName string) string {
```